### PR TITLE
Fix #175, CodeQL Action Workflow

### LINKED ISSUE #175: Add GitHub Actions CodeQL Workflow
**Is your feature request related to a problem? Please describe.**
The lgtm continuous integration checks don't always seem to run correctly. 

**Describe the solution you'd like**
This will apply the same technology, CodeQL, but using Github Actions workflows and unify our different CI systems. 

**Describe alternatives you've considered**
Keep using LGTM and figure out why we get so many "no code changes detected" messages.

**Additional context**
We can expand our CodeQL checks with ode standards and style checks once we update our code style and standard.

**Requester Info**
@astrogeco

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Describe the contribution**
Fix #175 
Implemented CodeQL Action as a workflow to automatically detect common vulnerabilities and coding errors. 

**Expected behavior changes**
Results should be displayed in the Security tab under Code Scanning Alerts. The workflow is triggered when code is pushed to any branch in a repository and during pull request events. 
image

**Additional context**
For more information visit: https://github.com/github/codeql-action

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

**Comment by astrogeco** [comment]
Do you have a sample of what our report looks like? I can't find it on your fork.

**Comment by ArielSAdamsNASA** [comment]
> Do you have a sample of what our report looks like? I can't find it on your fork. Is there a configuration file for CodeQL?

CodeQL [results ](https://github.com/ArielSAdamsNASA/cFS/security/code-scanning) will be found in the code scanning alerts section of the security tab. The configuration file for CodeQL is the [codeql-build-cfs.yml](https://github.com/ArielSAdamsNASA/cFS/blob/main/.github/workflows/codeql-build-cfs.yml) file.

**Comment by astrogeco** [comment]
Cool! 

I couldn't access your link but I was able to see the list at
https://github.com/nasa/cFS/security/code-scanning?query=ref%3Arefs%2Fpull%2F165%2Fmerge

**Comment by astrogeco** [comment]
For the configuration file, I meant something like this: https://github.com/github/codeql-action#configuration-file so we can specify which vulnerabilities to look for.

File: .github/workflows/codeql-build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: "CodeQL Analysis"
+
+on:
+  push:
```
Comment (astrogeco) [inline .github/workflows/codeql-build.yml:4]: See updated workflow files. This should now be 

``` yml
on:
  push:
    branches:
      - main
  pull_request:
```

Comment (ArielSAdamsNASA) [reply .github/workflows/codeql-build.yml:4]: Updated to push on main branch.

File: .github/workflows/codeql-build.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  ENABLE_UNIT_TESTS: true
+  OMIT_DEPRECATED: true
+
+jobs:
+
+  CodeQL-Build:
+
+    strategy:
+      matrix:
+        buildtype: [debug, release]
```
Comment (astrogeco) [inline .github/workflows/codeql-build.yml:20]: @skliper should we check security stuff for both debug and release builds?

**Comment by astrogeco** [comment]
Opened #175. Please Update commit format and PR title to reflect that this PR closes that issue.

**Comment by ArielSAdamsNASA** [comment]
> Opened #175. Please Update commit format and PR title to reflect that this PR closes that issue.

@astrogeco Completed.

**Review by skliper — APPROVED** [review]
Looks fine to me, but I haven't used CodeQL.

**Comment by astrogeco** [comment]
> Looks fine to me, but I haven't used CodeQL.

CodeQL is the engine behind LGTM so we're technically using it.

File: .github/workflows/codeql-build.yml (hunk 2 above)
Comment (skliper) [reply .github/workflows/codeql-build.yml:20]: release test would be sufficient in my mind... always with OMIT_DEPRECATED

**Comment by astrogeco** [comment]
@ArielSAdamsNASA can you add a screenshot of what the code report looks like. It might be useful for the CCB

**Comment by astrogeco** [comment]
[CCB:2021-01-21](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-January-21-2021) APPROVED

**Pull request merged**